consensus/src/pow/work.rs

```diff
@@ -0,0 +1,133 @@
+#![allow(dead_code)]
+
+use crate::pow::helpers::{
+    calculate_new_target, check_difficulty, is_for_retarget, retarget_block_time, special_rules,
+};
+use crate::pow::temp::BlockIndex;
+use crate::pow::{Error, PoW};
+use common::chain::block::Block;
+use common::primitives::consensus_data::{ConsensusData, PoWData};
+use common::primitives::{Compact, Idable};
+use common::Uint256;
+
+pub fn check_proof_of_work(hash: Uint256, bits: Compact) -> bool {
+    if let Ok(target) = Uint256::try_from(bits) {
+        hash > target
+    } else {
+        false
+    }
+}
+
+impl PoW {
+    /// The difference (in block time) between the current block and 2016th block before the current one.
+    /// This difference should be inclusive between (2 weeks/4) and (2 weeks*4).
+    /// See Bitcoin's Protocol rules on [Difficulty change](https://en.bitcoin.it/wiki/Protocol_rules)
+    pub fn actual_timespan(&self, prev_block_blocktime: u32, retarget_blocktime: u32) -> u64 {
+        let actual_timespan = (prev_block_blocktime - retarget_blocktime) as u64;
+
+        // 2 weeks / 4
```

Comment:
This is the wrong place for these comments. It should be where you put the numbers. Here you just explain the concept.

Reply:
Also, btw, this isn't 2 weeks in mintlayer. It's in bitcoin.